packages/fuels-programs/src/assembly/script_and_predicate_loader.rs

```diff
@@ -296,6 +321,38 @@ pub fn split_at_data_offset(binary: &[u8]) -> Result<(&[u8], &[u8])> {
             binary.len()
         ));
     }
-
     Ok(binary.split_at(offset))
 }
+
+pub fn split_binary(binary: &[u8]) -> Result<(&[u8], &[u8])> {
+    // First determine if it's a legacy binary
+    match is_legacy_binary(binary)? {
+        true => split_at_data_offset(binary),
+        false => split_at_configurable_offset(binary),
+    }
+}
+
+pub fn is_legacy_binary(binary: &[u8]) -> Result<bool> {
+    if binary.len() < 8 {
+        return Err(fuels_core::error!(
+            Other,
+            "binary too short to check JMPF instruction, need at least 8 bytes but got: {}",
+            binary.len()
+        ));
+    }
+
+    match (binary[4], binary[7]) {
+        (0x74, 0x02) => Ok(true),
```

Comment:
it would also be nice that the idea of the presence of 74 and 02 is not present in two places since, this is duplicated in the `is_legacy_check`.